numbat/src/bytecode_interpreter.rs

```diff
@@ -109,6 +109,32 @@ impl BytecodeInterpreter {
                 self.compile_expression(lhs)?;
                 self.vm.add_op(Op::LogicalNeg);
             }
+            Expression::BinaryOperator(
+                _span,
+                op @ (BinaryOperator::LogicalOr | BinaryOperator::LogicalAnd),
+                lhs,
+                rhs,
+                _type,
+            ) => {
+                self.compile_expression(lhs)?;
+
+                let jump_offset = self.vm.current_offset() + 1; // +1 for the opcode
+                self.vm.add_op1(
+                    match op {
+                        BinaryOperator::LogicalOr => Op::JumpIfTrue,
+                        BinaryOperator::LogicalAnd => Op::JumpIfFalse,
+                        _ => unreachable!(),
+                    },
+                    0xffff,
+                );
+                self.vm.add_op(Op::Pop);
+
+                self.compile_expression(rhs)?;
+
+                let else_block_offset = self.vm.current_offset();
+                self.vm
+                    .patch_u16_value_at(jump_offset, else_block_offset - (jump_offset + 2));
```

Comment:
This is a bit unclear to me. I think I follow, but a comment to explain what is happening would be helpful.